source/applySpec.js

```diff
@@ -9,19 +9,20 @@ import values from './values';
 
 
 /**
- * Given a spec object recursively mapping properties to functions, creates a
- * function producing an object of the same structure, by mapping each property
- * to the result of calling its associated function with the supplied arguments.
+ * Given a spec hierarchy of objects and/or arrays recursively mapping properties/elements
+ * to functions, creates a value of the same structure, by mapping each
+ * property/element to the result of calling its associated function with
+ * the supplied arguments.
  *
  * @func
  * @memberOf R
  * @since v0.20.0
  * @category Function
- * @sig {k: ((a, b, ..., m) -> v)} -> ((a, b, ..., m) -> {k: v})
- * @param {Object} spec an object recursively mapping properties to functions for
- *        producing the values for these properties.
- * @return {Function} A function that returns an object of the same structure
- * as `spec', with each property set to the value returned by calling its
+ * @sig {k: ((a, b, ..., m) -> v)} | [((a, b, ..., m) -> v)] -> ((a, b, ..., m) -> {k: v} | [v])
```

Comment:
I'm not really sure how to write a good signature for this with its recursive nature as well as the variadic functions.  But this one simply confuses me.

I don't know if we're better with a list of related signatures, but they would at least individually be much simpler:

```hs
applySpec :: {k: a -> v} -> (a -> {k: v}
          :: {k: (a, b) -> v} -> ((a, b) -> {k, v})
          :: {k: (a, b, c) -> v} -> ((a, b, c) -> {k, v})
          :: ...
          :: {[(a -> v)] -> (a -> [v]) 
          :: {[((a, b) -> v)] -> ((a, b) -> [v]) 
          :: {[((a, b, c) -> v)] -> ((a, b, c) -> [v]) 
          :: ...
```

But that still does not capture the recursive nature of this.  I now very much regret that this is recursive; we seem to have abandoned our focus on simplicity here.